### PR TITLE
Replace menuinst subprocessing by ctypes win elevation

### PR DESCRIPTION
**Pull request by goanpeca**

### CODE REVIEW

File: conda/common/os/windows.py [hunk 1]
```diff
@@ -38,3 +71,25 @@ def is_admin_on_windows():  # pragma: unix no cover
...
+
+def run_as_admin(cmd_line):
+    """
+    See:
+    - http://stackoverflow.com/a/19719292/1170370 on 20160407 MCS.
+    - msdn.microsoft.com/en-us/library/windows/desktop/bb762153(v=vs.85).aspx
+    """
+    code = None
+    if _ctypes:
+        params = " ".join(['"%s"' % (x, ) for x in cmd_line[1:]])
```
Comment (kalefranz) [inline conda/common/os/windows.py]: This line looks off to me.  Is it really ok to quote every single parameter?  And why the single-value tuple for the string interpolation?

Comment (goanpeca) [reply conda/common/os/windows.py]: > Is it really ok to quote every single parameter?

Well we could do it more conditionally based on the contents, but I think it is ok.

**Comment by kalefranz** [comment]
```
INFO conda.core.initialize:run_plan(487): init_cmd_exe_registry: FileNotFoundError(2, 'The system cannot find the file specified', None, 2, None)
Traceback (most recent call last):
  File "C:\Users\admin\conda\conda\core\initialize.py", line 485, in run_plan
    result = globals()[step['function']](*step.get('args', ()), **step.get('kwargs', {}))
  File "C:\Users\admin\conda\conda\core\initialize.py", line 1064, in init_cmd_exe_registry
    prev_value, value_type = _read_windows_registry(target_path)
  File "C:\Users\admin\conda\conda\core\initialize.py", line 1036, in _read_windows_registry
    value_tuple = winreg.QueryValueEx(key, value_name)
FileNotFoundError: [WinError 2] The system cannot find the file specified
```

**Comment by goanpeca** [comment]
@kalefranz not sure about the error, as I have not change any code related to that. 

Could you test on your VM?

File: conda/cli/main_init.py [hunk 2]
```diff
@@ -48,6 +48,5 @@ def execute(args, parser):
             for_user = False
 
         anaconda_prompt = on_win and args.anaconda_prompt
-        anaconda_prompt = on_win  # TODO: probably remove and leave --anaconda-prompt as a flag
```
Comment (kalefranz) [inline conda/cli/main_init.py]: I just have put that in just for debug or something 😳

File: conda/core/initialize.py [hunk 3]
```diff
@@ -537,7 +532,7 @@ def run_plan_elevated(plan):
         else:
             stdin = json.dumps(plan)
             result = subprocess_call(
-                'sudo %s -m conda.initialize' % sys.executable,
+                'sudo %s -m conda.core.initialize' % sys.executable,
```
Comment (kalefranz) [inline conda/core/initialize.py]: No wonder it wasn't working!

Comment (goanpeca) [reply conda/core/initialize.py]: yep I told you it was a silly bug 🤣

**Comment by goanpeca** [comment]
```
=================================== FAILURES ===================================
__________ IntegrationTests.test_install_update_deps_only_deps_flags ___________
Traceback (most recent call last):
  File "/home/circleci/conda/tests/test_create.py", line 838, in test_install_update_deps_only_deps_flags
    raise NotImplementedError()
NotImplementedError
=============================== warnings summary ===============================
None
  pytest-catchlog plugin has been merged into the core, please remove it from your requirements.

-- Docs: http://doc.pytest.org/en/latest/warnings.html
============================= 492 tests deselected =============================
 1 failed, 113 passed, 28 skipped, 492 deselected, 2 xfailed, 1 warnings in 1707.25 seconds

**Comment by kalefranz** [comment]
I added a merge commit to this PR to merge in master.  Want to see the tests run.

**Comment by goanpeca** [comment]
@kalefranz tests pass now

**Comment by kalefranz** [comment]
Was there anything else you wanted to do here?  Didn't you want to try to finish something out that you sort of started thinking through?  Is that done, or not as high priority now?  I don't remember exactly what that was...

**Comment by goanpeca** [comment]
I could not make it work ( reattach the elevated process to the current condole to view anyboutput)

**Pull request merged**